dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/configuration/JobParametersConfiguration.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.configuration;
+
+
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import uk.ac.ebi.eva.dbsnpimporter.parameters.DbsnpDatasource;
+import uk.ac.ebi.eva.dbsnpimporter.parameters.Parameters;
+
+import java.util.HashMap;
+
+@Configuration
+public class JobParametersConfiguration {
+
+    private HashMap<String, JobParameter> parametersMap;
+
+    @Bean
+    JobParameters jobParameters(Parameters parameters, DbsnpDatasource dbsnpDatasource,
+                                MongoProperties mongoProperties) {
+
+        parametersMap = new HashMap<>();
+
+        addParameter("assembly", parameters.getAssembly());
+        addParameter("assemblyTypes", String.join(",", parameters.getAssemblyTypes()));
+        addParameter("batchId", parameters.getBatchId());
+        addParameter("chunkSize", parameters.getChunkSize());
+        addParameter("dbsnpBuild", parameters.getDbsnpBuild());
+        addParameter("pageSize", parameters.getPageSize());
+        addParameter("processor", parameters.getProcessor());
+        addParameter("variantsCollection", parameters.getVariantsCollection());
+
+        addParameter("driverClassName", dbsnpDatasource.getDriverClassName());
+        addParameter("url", dbsnpDatasource.getUrl());
+        addParameter("username", dbsnpDatasource.getUsername());
+        // NOTE: not putting the password on purpose. is it safe to put a readonly password in the jobRepository?
+
+        addParameter("mongoAuthenticationDatabase", mongoProperties.getAuthenticationDatabase());
+        addParameter("mongoDatabase", mongoProperties.getDatabase());
+        addParameter("mongoHost", mongoProperties.getHost());
+        addParameter("mongoPort", mongoProperties.getPort());
+        addParameter("mongoUri", mongoProperties.getUri());
+        addParameter("mongoUsername", mongoProperties.getUsername());
+        // NOTE: not putting the password on purpose. is it safe to put a write password in the jobRepository?
+
+        return new JobParameters(parametersMap);
```

Comment:
We may want to use a [JobParametersBuilder](https://docs.spring.io/spring-batch/trunk/apidocs/org/springframework/batch/core/JobParametersBuilder.html) to define which parameters are identifying for a job. For instance, I don't think the Mongo credentials should be identifying.